stdlib/test/collections/test_counter.mojo

```diff
@@ -32,6 +32,206 @@ def test_counter_getitem():
     assert_equal(c[5], 0)
 
 
+def test_iter():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+
+    var keys = String("")
+    for key in c:
+        keys += key[]
+
+    assert_equal(keys, "ab")
+
+
+def test_iter_keys():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+
+    var keys = String("")
+    for key in c.keys():
+        keys += key[]
+
+    assert_equal(keys, "ab")
+
+
+def test_iter_values():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+
+    var sum = 0
+    for value in c.values():
+        sum += value[]
+
+    assert_equal(sum, 3)
+
+
+def test_iter_values_mut():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+
+    for value in c.values():
+        value[] += 1
+
+    assert_equal(2, c["a"])
+    assert_equal(3, c["b"])
+    assert_equal(2, len(c))
+
+
+def test_iter_items():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+
+    var keys = String("")
+    var sum = 0
+    for entry in c.items():
+        keys += entry[].key
+        sum += entry[].value
+
+    assert_equal(keys, "ab")
+    assert_equal(sum, 3)
+
+
+def test_total():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+
+    assert_equal(c.total(), 3)
+
+
+def test_most_common():
+    var c = Counter[String]()
+    c["a"] = 1
+    c["b"] = 2
+    c["c"] = 3
+
+    var most_common = c.most_common(2)
+    assert_equal(len(most_common), 2)
+    assert_equal(most_common[0][0][String], "c")
+    assert_equal(most_common[0][1][Int], 3)
+    assert_equal(most_common[1][0][String], "b")
+    assert_equal(most_common[1][1][Int], 2)
+
+
+def test_eq_and_ne():
+    var c1 = Counter[String]()
+    c1["a"] = 1
+    c1["b"] = 2
+    c1["d"] = 0
+
+    var c2 = Counter[String]()
+    c2["a"] = 1
+    c2["b"] = 2
+    c2["c"] = 0
+
+    assert_true(c1 == c2)
+    assert_false(c1 != c2)
+
+    c2["b"] = 3
+    assert_false(c1 == c2)
+    assert_true(c1 != c2)
+
+
+def test_lt_le_gt_and_ge():
+    var c1 = Counter[String]()
+    c1["a"] = 1
+    c1["b"] = 2
+    c1["d"] = 0
+
+    var c2 = Counter[String]()
+    c2["a"] = 1
+    c2["b"] = 2
+    c2["c"] = 0
+
+    assert_false(c1 < c2)
+    assert_true(c1 <= c2)
+    assert_false(c1 > c2)
+
+    c2["b"] = 3
+    assert_true(c1 < c2)
+    assert_true(c1 <= c2)
```

Comment:
Looks like `>=` is untested. Could you please add a cast for that?

Reply:
Also for this and the other comparison methods, could you please test the dunder methods directly, instead of using the operators?

Reply:
Done: https://github.com/msaelices/mojo/commit/dc2d2079c9a39764fc585103bdf70bd923e103a3